node/full_node_integration_test.go

```diff
@@ -482,16 +481,20 @@ func TestSubmitBlocksToDA(t *testing.T) {
 		assert.NoError(seq.Stop())
 	}()
 
-	timer := time.NewTimer(5 * seq.nodeConfig.DABlockTime)
-	<-timer.C
-
 	numberOfBlocksToSyncTill := seq.Store.Height()
 
 	//Make sure all produced blocks made it to DA
 	for i := uint64(1); i <= numberOfBlocksToSyncTill; i++ {
-		block, err := seq.Store.GetBlock(i)
-		require.NoError(err)
-		require.True(seq.blockManager.IsDAIncluded(block.Hash()), block.Height())
+		require.NoError(testutils.Retry(300, 100*time.Millisecond, func() error {
+			block, err := seq.Store.GetBlock(i)
+			if err != nil {
+				return err
+			}
+			if !seq.blockManager.IsDAIncluded(block.Hash()) {
+				return fmt.Errorf("block %d not DA included", block.Height())
+			}
+			return nil
+		}))
```

Comment:
replacing the timer, which is essentially a sleep, with a dynamic retry makes the test more robust.

Reply:
Can be extracted in a helper